stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -1236,36 +410,65 @@ public class MessageListViewModel(
+
+    /**
+     * Sets a handler which determines the position of a message inside a group.
+     *
+     * @param messagePositionHandler The handler to use.
+     */
+    public fun setMessagePositionHandler(messagePositionHandler: MessagePositionHandler) {
+        messageListController.setMessagePositionHandler(messagePositionHandler)
+    }
+
+    /**
+     * Sets the date separator handler which determines when to add date separators.
+     * By default, a date separator will be added if the difference between two messages' dates is greater than 4h.
+     *
+     * @param dateSeparatorHandler The handler to use. If null, the messages list won't contain date separators.
+     */
+    public fun setDateSeparatorHandler(dateSeparatorHandler: DateSeparatorHandler?) {
+        messageListController.setDateSeparatorHandler(dateSeparatorHandler)
     }
 
-    internal companion object {
-        /**
-         * The default threshold for showing date separators. If the message difference in hours is equal to this
-         * number, then we show a separator, if it's enabled in the list.
-         */
-        internal const val DateSeparatorDefaultHourThreshold: Long = 4
+    /**
+     * Sets thread date separator handler which determines when to add date separators inside the thread.
+     *
+     * @param threadDateSeparatorHandler The handler to use. If null, the thread messages list won't contain date
+     * separators.
+     */
+    public fun setThreadDateSeparatorHandler(threadDateSeparatorHandler: DateSeparatorHandler?) {
+        messageListController.setThreadDateSeparatorHandler(threadDateSeparatorHandler)
+    }
+
+    /**
+     * Sets the value used to determine if message footer content is shown.
+     * @see MessageFooterVisibility
+     *
+     * @param messageFooterVisibility Changes the visibility of message footers.
+     */
+    public fun setMessageFooterVisibility(messageFooterVisibility: MessageFooterVisibility) {
+        messageListController.setMessageFooterVisibility(messageFooterVisibility)
+    }
 
-        /**
-         * The default limit for messages count in requests.
-         */
-        internal const val DefaultMessageLimit: Int = 30
+    /**
+     * Sets the value used to filter deleted messages.
+     * @see DeletedMessageVisibility
+     *
+     * @param deletedMessageVisibility Changes the visibility of deleted messages.
+     */
+    public fun setDeletedMessageVisibility(deletedMessageVisibility: DeletedMessageVisibility) {
+        messageListController.setDeletedMessageVisibility(deletedMessageVisibility)
+    }
 
-        /**
-         * Time in millis, after which the focus is removed.
-         */
-        private const val RemoveMessageFocusDelay: Long = 2000
+    /**
+     * Sets whether the system messages should be visible.
+     *
+     * @param showSystemMessages Whether system messages should be visible or not.
+     */
+    public fun setAreSystemMessagesVisible(showSystemMessages: Boolean) {
```

Comment:
Same as the other two comments mentioning that if we adopt the previous naming suggestion made about the controller counterpart of this function, we should adopt it here as well.